Pylint is not catching all errors when running via Github workflow

### Steps to Reproduce

Run the following:
```
docker-compose run web pylint ./**/*.py
```

It will detect some errors.  These errors are not reported when pylint runs via github workflow.

on my dev env running the above command also returns a bunch of errors we probably _dont_ care about, like in migrations and whatnot. Here's the output I get when I run it on the main branch for instance:

```
************* Module conftest
conftest.py:6:0: W0401: Wildcard import fixtures.common (wildcard-import)
conftest.py:6:0: W0614: Unused import drf_client from wildcard import (unused-wildcard-import)
conftest.py:6:0: W0614: Unused import APIClient from wildcard import (unused-wildcard-import)
************* Module content_sync.apis.github
content_sync/apis/github.py:47:0: C0116: Missing function or method docstring (missing-function-docstring)
content_sync/apis/github.py:78:0: R0914: Too many local variables (21/15) (too-many-locals)
content_sync/apis/github.py:123:8: W0612: Unused variable 'ye' (unused-variable)
************* Module content_sync.backends.base_test
content_sync/backends/base_test.py:59:8: E0110: Abstract class '_NotImplementedBackend' with abstract methods instantiated (abstract-class-instantiated)
************* Module content_sync.migrations.0001_add_content_sync_state
content_sync/migrations/0001_add_content_sync_state.py:1:0: C0114: Missing module docstring (missing-module-docstring)
content_sync/migrations/0001_add_content_sync_state.py:7:0: C0115: Missing class docstring (missing-class-docstring)
************* Module content_sync.migrations.0002_backpopulate_content_sync_state
content_sync/migrations/0002_backpopulate_content_sync_state.py:1:0: C0114: Missing module docstring (missing-module-docstring)
content_sync/migrations/0002_backpopulate_content_sync_state.py:17:43: W0613: Unused argument 'schema_editor' (unused-argument)
content_sync/migrations/0002_backpopulate_content_sync_state.py:35:0: C0115: Missing class docstring (missing-class-docstring)
************* Module content_sync.migrations.0003_content_sync_state_nullable_content
content_sync/migrations/0003_content_sync_state_nullable_content.py:1:0: C0114: Missing module docstring (missing-module-docstring)
content_sync/migrations/0003_content_sync_state_nullable_content.py:7:0: C0115: Missing class docstring (missing-class-docstring)
************* Module content_sync.migrations.0004_content_not_nullable
content_sync/migrations/0004_content_not_nullable.py:1:0: C0114: Missing module docstring (missing-module-docstring)
content_sync/migrations/0004_content_not_nullable.py:7:30: W0613: Unused argument 'schema_editor' (unused-argument)
content_sync/migrations/0004_content_not_nullable.py:13:0: C0115: Missing class docstring (missing-class-docstring)
************* Module 0001_create_default_robots_txt
main/migrations/0001_create_default_robots_txt.py:1:0: C0114: Missing module docstring (missing-module-docstring)
main/migrations/0001_create_default_robots_txt.py:7:0: C0116: Missing function or method docstring (missing-function-docstring)
main/migrations/0001_create_default_robots_txt.py:7:36: W0613: Unused argument 'schema_editor' (unused-argument)
main/migrations/0001_create_default_robots_txt.py:30:0: C0115: Missing class docstring (missing-class-docstring)
************* Module railroad_diagrams
node_modules/railroad-diagrams/railroad_diagrams.py:1:0: C0114: Missing module docstring (missing-module-docstring)
node_modules/railroad-diagrams/railroad_diagrams.py:12:0: C0116: Missing function or method docstring (missing-function-docstring)
node_modules/railroad-diagrams/railroad_diagrams.py:15:0: C0116: Missing function or method docstring (missing-function-docstring)
node_modules/railroad-diagrams/railroad_diagrams.py:26:0: C0115: Missing class docstring (missing-class-docstring)
node_modules/railroad-diagrams/railroad_diagrams.py:26:0: R0205: Class 'DiagramItem' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)

-----------------------------------
Your code has been rated at 9.67/10
```

(I truncated some of the output)

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -62,7 +62,7 @@ jobs:
         run: pip install -r requirements.txt -r test_requirements.txt
 
       - name: Lint
-        run: pylint ./**/*.py
+        run: pwd | xargs pylint
         env:
           PYTHONPATH: "."
 

--- a/conftest.py
+++ b/conftest.py
@@ -3,7 +3,7 @@ from types import SimpleNamespace
 
 import pytest
 
-from fixtures.common import *
+from fixtures.common import *  # pylint:disable=wildcard-import,unused-wildcard-import
 from websites.constants import COURSE_STARTER_SLUG
 from websites.models import WebsiteStarter
 

--- a/content_sync/apis/github.py
+++ b/content_sync/apis/github.py
@@ -45,6 +45,9 @@ class SyncResult:
 
 
 def decode_file_contents(content_file: ContentFile) -> str:
+    """
+    Decode repo file contents from base64 to a normal string.
+    """
     return str(b64decode(content_file.content), encoding="utf-8")
 
 
@@ -75,7 +78,7 @@ def get_destination_filepath(
     return None
 
 
-def sync_starter_configs(
+def sync_starter_configs(  # pylint:disable=too-many-locals
     repo_url: str, config_files: List[str], commit: Optional[str] = None
 ):
     """
@@ -110,7 +113,7 @@ def sync_starter_configs(
             if created:
                 starter.name = starter.slug
                 starter.save()
-        except YamaleError as ye:
+        except YamaleError:
             log.exception("Invalid site config YAML found in %s", config_file)
             continue
         except:  # pylint: disable=bare-except

--- a/content_sync/backends/base_test.py
+++ b/content_sync/backends/base_test.py
@@ -55,10 +55,10 @@ def test_base_sync_backend_subclass_implemented(mocker):
 
 def test_base_sync_backend_subclass_not_implemented(mocker):
     """ Verify BaseSyncBackend doesn't unimplemented subclasses """
-    with pytest.raises(TypeError):
+    with pytest.raises(TypeError):  # pylint: disable=abstract-class-instantiated
         _NotImplementedBackend(
             mocker.Mock()
-        )  # pylint: disable=abstract-class-instantiated
+        )
 
 
 def test_sync_content_to_backend_create(mocker):

--- a/content_sync/backends/base_test.py
+++ b/content_sync/backends/base_test.py
@@ -55,8 +55,8 @@ def test_base_sync_backend_subclass_implemented(mocker):
 
 def test_base_sync_backend_subclass_not_implemented(mocker):
     """ Verify BaseSyncBackend doesn't unimplemented subclasses """
-    with pytest.raises(TypeError):  # pylint: disable=abstract-class-instantiated
-        _NotImplementedBackend(
+    with pytest.raises(TypeError):
+        _NotImplementedBackend(  # pylint: disable=abstract-class-instantiated
             mocker.Mock()
         )
 

--- a/websites/config_schema/api_test.py
+++ b/websites/config_schema/api_test.py
@@ -10,6 +10,9 @@ from websites.config_schema.api import (
 )
 
 
+# pylint:disable=redefined-outer-name
+
+
 SCHEMA_RESOURCES_DIR = "localdev/configs/"
 SCHEMA_CONFIG_FILE = "ocw-course-site-config.yml"
 VALID_TITLE_FIELD = {

--- a/websites/config_schema/validators.py
+++ b/websites/config_schema/validators.py
@@ -9,6 +9,10 @@ from websites.site_config_api import SiteConfig
 
 
 class AddedSchemaRule:
+    """
+    Class for validating site config schema rules
+    """
+
     @staticmethod
     def apply_rule(data):
         """
@@ -65,7 +69,7 @@ class UniqueNamesRule(AddedSchemaRule):
     @staticmethod
     def apply_rule(data):
         name_paths = defaultdict(list)
-        for i, config_item in enumerate(SiteConfig(data).iter_items()):
+        for _, config_item in enumerate(SiteConfig(data).iter_items()):
             name_paths[config_item.item["name"]].append(config_item.path)
         faulty_name_paths = {
             name: paths for name, paths in name_paths.items() if len(paths) > 1
@@ -94,7 +98,7 @@ class ContentFolderRule(AddedSchemaRule):
     def apply_rule(data):
         faulty_paths = {}
         site_config = SiteConfig(data)
-        for i, config_item in enumerate(site_config.iter_items()):
+        for _, config_item in enumerate(site_config.iter_items()):
             if config_item.is_folder_item() and not site_config.is_page_content(
                 config_item
             ):
@@ -123,7 +127,7 @@ class RequiredTitleRule(AddedSchemaRule):
     def apply_rule(data):
         faulty_paths = {}
         site_config = SiteConfig(data)
-        for i, config_item in enumerate(site_config.iter_items()):
+        for _, config_item in enumerate(site_config.iter_items()):
             title_field = first_or_none(
                 [field for field in config_item.fields if field["name"] == "title"]
             )

